qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -0,0 +1,424 @@
+        can be estimated with this experiment.
+
+        This experiment is performed by stretching the pulse duration of the cross resonance tone
+        and measuring the target qubit by projecting onto x, y, and z basis.
+        We will observe the control qubit state dependent (controlled-) Rabi oscillation on the
+        target qubit. The fit for this oscillation tomographically reconstructs the
+        Hamiltonian in the form shown above.
+        Thus the control qubit is prepared in both ground and excited state and
+        the experiment is repeated for both control qubit states.
+        See Ref. [1] for more details.
+
+        More specifically, the following circuits are executed in this experiment.
+
+        .. parsed-literal::
+
+            (X measurement)
+
+                 ┌───┐┌────────────────────┐
+            q_0: ┤ X ├┤0                   ├────────
+                 └───┘│  cr_tone(duration) │┌───┐┌─┐
+            q_1: ─────┤1                   ├┤ H ├┤M├
+                      └────────────────────┘└───┘└╥┘
+            c: 1/═════════════════════════════════╩═
+                                                  0
+
+            (Y measurement)
+
+                 ┌───┐┌────────────────────┐
+            q_0: ┤ X ├┤0                   ├───────────────
+                 └───┘│  cr_tone(duration) │┌─────┐┌───┐┌─┐
+            q_1: ─────┤1                   ├┤ Sdg ├┤ H ├┤M├
+                      └────────────────────┘└─────┘└───┘└╥┘
+            c: 1/════════════════════════════════════════╩═
+                                                         0
+
+            (Z measurement)
+
+                 ┌───┐┌────────────────────┐
+            q_0: ┤ X ├┤0                   ├───
+                 └───┘│  cr_tone(duration) │┌─┐
+            q_1: ─────┤1                   ├┤M├
+                      └────────────────────┘└╥┘
+            c: 1/════════════════════════════╩═
+                                             0
+
+        The ``X`` gate on the control qubit (``q_0``) depends on the required control state.
+        Here ``cr_tone`` is implemented by a single cross resonance tone
+        driving the control qubit at the frequency of target qubit.
+        The pulse envelope is the flat-topped Gaussian implemented by the parametric pulse
+        :py:class:`~qiskit.pulse.library.parametric_pulses.GaussianSquare`.
+        The effect of pulse edges are also considered as a net cross resonance pulse duration.
+        The net duration of pulse edges are approximated by
+
+        .. math::
+
+            \tau_{\rm edges} = \sqrt{2 \pi \sigma^2},
+
+        where the :math:`\sigma` is the sigma of Gaussian rising and falling edges.
+        Thus the net duration of the entire cross resonance pulse will become
+
+        .. math::
+
+            \tau = {\rm duration} - 2 r \sigma + \tau_{\rm edges},
+
+        where :math:`r` is the ratio of the actual edge duration to the sigma.
```

Comment:
So far, we have been fitting with an arbitrary offset. This is a nice approximation for the edge effects, though I think `r` ends up being equivalent to the arbitrary offset. Why not parametrize it as `\sqrt{2 \pi} \sigma r` so that `r` is a correction to sigma?

Reply:
Thanks Will. I feel that is the right direction. Could you please check update in https://github.com/Qiskit/qiskit-experiments/pull/385/commits/b1d7099db4e6fed46d8317f6d4a869699e4f46c9?

Basically in the new code we scan flat-top width rather than pulse duration, and edge effect is considered as an offset to scan value (xval), which is also an extra fit parameter with initial guess of `\sqrt{2 \pi} \sigma`.